### PR TITLE
[ENHANCEMENT / BUGFIX] CI: publish version docs on tag creation

### PR DESCRIPTION
**Pull request by frascuchon**
This PR reviews the specific tag version deployment and deploys (and pushes) versions when a tag in the format vX.Y.Z is pushed. 

- The `--push` option was missing in the previous version
- The version deployed will keep only the minor version. That means tags for a patch version `v2.1.3` will deploy the docs `v2.1`

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- follows the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: .github/workflows/argilla.docs.yml [hunk 1]
```diff
@@ -62,5 +62,11 @@ jobs:
       - run: pdm run mike deploy dev --push
         if: github.ref == 'refs/heads/develop'
 
-      - run: pdm run mike deploy ${{ github.ref_name }}
+      - name: pdm run mike deploy $version --push
```
Comment (davidberenstein1957) [inline .github/workflows/argilla.docs.yml]: Perhaps we can add more specific names to the other steps too?

File: .github/workflows/argilla.docs.yml [hunk 2]
```diff
@@ -62,5 +62,11 @@ jobs:
...
+        run: |
+          version=$(echo $TAG_VERSION | awk -F \. {'print $1"."$2'})
+          echo "Deploying version ${version}"
+          pdm run mike deploy $version --push
         if: startsWith(github.ref, 'refs/tags/')
+        env:
+          TAG_VERSION: ${{ github.ref_name }}
```
Comment (davidberenstein1957) [inline .github/workflows/argilla.docs.yml:75]: This is used internally?

**Comment by davidberenstein1957** [comment]
@frascuchon my comment were resolved #5088

**Comment by frascuchon** [comment]
> @frascuchon my comment were resolved #5088

Nice. I think we can merge this on once #5088 is merged

**Pull request merged**